usd_qtpy/lib/usd.py

```diff
@@ -320,3 +321,28 @@ def remove_spec(spec):
 
     else:
         raise TypeError(f"Unsupported spec type: {spec}")
+
+
+def traverse_prim_descendants(stage: Usd.Stage, path: Union[Sdf.Path,str]) -> list[Usd.Prim]:
+    """
+    DFS Iterate through hierarchy of primitive at path.
+    """
+    if isinstance(path,str):
+        path = Sdf.Path(path)
+
+    prim = stage.GetPrimAtPath(path)
+
+    def recurse_children(prim: Usd.Prim, carry: list[Usd.Prim] = None) -> list[Usd.Prim]:
+        if carry is None:
+            carry = []
+        
+        children: list[Usd.Prim] = prim.GetAllChildren()
+        if children:
+            carry.extend(children)
+
+            for child in children:
+                carry = recurse_children(child,carry)
+        
+        return carry
+    
```

Comment:
Why do you need this? And how does this differ from `Usd.Stage.Traverse`? :)

Reply:
Or if you want to start from a particular prim, `Usd.PrimRange(prim)`?

Reply:
> Why do you need this? And how does this differ from `Usd.Stage.Traverse`? :)
> Or if you want to start from a particular prim, `Usd.PrimRange(prim)`?

I thought I needed it to safely reparent all primitives, but that turns out to not be the case.
Rebasing is all I need!

And yes this is EXACTLY the same as PrimRange down to the depth-first traversal, I just didn't know it existed yet, my bad hahaha. Will take it out again. next commit.